### PR TITLE
[WIP] Fix IDN Issues 

### PR DESCRIPTION
**Pull request by jaapmarcus**
I think is the right way or do we need to convert all vars to domain_idn and do the same changes?

### CODE REVIEW

**Review by Lupul — CHANGES_REQUESTED** [review]
Looks good, only minor changes needed

File: bin/v-add-web-domain-alias [hunk 1]
```diff
@@ -12,9 +12,8 @@
 
 # Argument definition
 user=$1
-domain=$2
-domain_idn=$2
-aliases=$3
+domain=$(idn -t --quiet -u "$2" )
+aliases=$(idn -t --quiet -u "$3" )
```
[the remark this answers is not in the document]
Comment (jaapmarcus) [reply bin/v-add-web-domain-alias]: Checked /web/edit/web/ 
https://github.com/hestiacp/hestiacp/blob/2e05ab4f56d66f94e26cec430c6adccecd35e780/web/edit/web/index.php#L245

[the remark this answers is not in the document]
Comment (jaapmarcus) [reply bin/v-add-web-domain-alias]: Synced behaviour with https://github.com/hestiacp/hestiacp/blob/main/bin/v-delete-web-domain-alias.

File: bin/v-add-web-domain [hunk 2]
```diff
@@ -127,6 +126,16 @@ else
             ALIAS="www.$domain"
         fi
     else
+        aliases_list=$(echo $aliases | tr "," "\n")
+        aliases=''
+        for alias in $aliases_list
+        do
+            if [ -e $aliases ]; then
```
[the remark this answers is not in the document]
Comment (jaapmarcus) [reply bin/v-add-web-domain]: Changed with commit

**Comment by mindaugasdi** [comment]
I created IDN alias for existing website and all my domains dissapeared (it says "0 web domains").
I hope this commit will solve the issue, but I wonder what I will do until then.

Other symptoms:
- Website itself continue to work.
- Newly "added" alias is non-working.
- It does not allow to edit: if going back history, ir redirects to empty domain list, possibly because old token is no longer valid.

**Comment by mindaugasdi** [comment]
I wrote this command below and domains are now visible again :tada:
`/usr/local/hestia/bin/v-delete-web-domain-alias {user} {domain} {alias}`

**Comment by mindaugasdi** [comment]
Creating alias in machine format worked second time for some reason :)
But if creating mail on IDN domain in machine format, it is not working (ssl do not apply, roundcube do not open). Creating mail on domain without IDN works in contrast.
**Note.** My comments are for stable hestia version without this update.

**Comment by jaapmarcus** [comment]
Thank you for your input. Are you able to test those changes on an test server?

**Comment by mindaugasdi** [comment]
I will try to test on the live server :flushed:
Since testing on test I am not prepared.

**Comment by mindaugasdi** [comment]
Test results
1. Tried to create domain (once in raw format, and once again in machine format) on the frontend and it created empty '' domain both times.
2. I was able to delete empty domain using CLI with some errors. I hope where was no side-effects because of this, on my live server :D

```
sudo /usr/local/hestia/bin/v-delete-mail-domain '{username}' ''

rm: cannot remove '/etc/exim4/domains/': Is a directory
rm: cannot remove '/home/{username}/conf/mail/': Permission denied
Error: mail domain yes doesn't exist
```

3. Tried to create domain using CLI (in machine format only this time). It created domain, printed errors bellow and **Hestia Frontend does not list this domain**. It does display domain count, but empty list. The same way empty as I described in my first comment week ago when using stable version. So, things on the frontend also needs to be changed.

```
sudo /usr/local/hestia/bin/v-add-mail-domain '{username}' '{idn_domain_machine_format}' 'yes' 'yes' 'yes'

mkdir: cannot create directory ‘/home/{username}/mail/{idn_domain_machine_format}‘: No such file or directory
chmod: cannot access '/home/{username}/mail/{idn_domain_machine_format}': No such file or directory
chown: cannot access '/home/{username}/mail/{idn_domain_machine_format}': No such file or directory
```

4. CLI correctly list this domain this time (raw translated format)

```
{
    "šššššššššš.td": {
        "ANTIVIRUS": "yes",
        "ANTISPAM": "yes",
        "DKIM": "yes",
        "CATCHALL": "",
        "ACCOUNTS": "0",
        "U_DISK": "0",
        "SSL": "no",
        "SUSPENDED": "no",
        "TIME": "16:17:21",
        "DATE": "2020-09-30"
    }
}
```

5. Since domain name is not empty this time, I can delete domain using CLI with no errors.

**Comment by jaapmarcus** [comment]
This current system will not work we should convert them to $(idn -t --quiet -a "$2" ) and then it will work probally find. How ever Don't have an IDN domain my self and would really like do some testing how ever don't want to spend 20 usd for a domain a will never use anymore

**Comment by mindaugasdi** [comment]
You could find it cheaper :) For example, I wrote "foömff.ch" to hostinger.com and it costs $8.68 with taxes included.
Eventually somehow IDN domain worked for me for website, but not email.

**Pull request closed without merging**